common/main/java/com/couchbase/lite/AbstractQuery.java

```diff
@@ -190,36 +205,17 @@ public ListenerToken addChangeListener(@Nullable Executor executor, @NonNull Que
     @Override
     public void removeChangeListener(@NonNull ListenerToken token) {
         Preconditions.assertNotNull(token, "token");
-        getLiveQuery().removeChangeListener(token);
+        final C4QueryObserver observer = listeners.remove(token);
+        if (observer != null) { observer.close(); }
     }
 
-    @NonNull
-    public ListenerToken newAddChangeListener(@Nullable Executor executor, @NonNull QueryChangeListener listener) {
-        Preconditions.assertNotNull(listener, "listener");
-
-        final ChangeListenerToken<QueryChange> token = new ChangeListenerToken<>(executor, listener);
-        final C4QueryObserver queryObserver;
-        try { queryObserver = C4QueryObserver.create(getC4QueryLocked(), token, this::onQueryChanged); }
+    private C4QueryObserver registerC4QueryObserver(ChangeListenerToken token) {
```

Comment:
I don't understand this change at all.  What is it that it accomplishes?

Reply:
In your code you said you can't throw the Exception because of API change, would this solve that?